client/src/components/messenger/Chat.js

```diff
@@ -1,51 +1,94 @@
 import { useState, useEffect, React } from "react";
 import { useSelector } from "react-redux";
 import { ChatController } from "./controllers/Chat";
+import { Alert } from "../AlertDialog";
 
 
 const Chat = ({ conversationId, title }) => {
     const [messages, setMessages] = useState([]);
-    const { userId } = useSelector((state) => state.user.userInfo);
+    const [text, setText] = useState("");
+    const [updater, setUpdater] = useState(null);
+    const alertDialog = Alert.useAlertDialog();
+
     // Get the token from the users global state.
     const token = useSelector((state) => state.user.token);
+    const { id } = useSelector((state) => state.user.userInfo);
+
+    const handleText = (e) => {
+        if (conversationId) // if a conversation is not active, disable text box
+            setText(e.target.value);
+    };
+
+    const handleSend = async () => {
+        // if a conversation is not active, disable send button
+        if (!conversationId)
+            return;
+        // if text box is empty dont bother trying to send message
+        if (text?.length < 1)
+            return;
+        let result = await ChatController.sendMessage(token, conversationId, text);
+        if (result.success) {
+            setText(""); // clear the text box
+            refresh(); // refresh the chat immediately to see the newly sent message
+        } else // display error message
+            alertDialog.display({
+                title: "Error",
+                message: result.msg,
+                open: true
+            });
+    };
 
-    useEffect(() => { // will call this at a defined interval later
-        const runAsync = async () => setMessages(await ChatController.getMessages(token, conversationId));
-        runAsync();
+    const refresh = async () => {
+        setMessages((await ChatController.getMessages(token, conversationId)).reverse());
+    };
+
+    useEffect(() => {
+        // refresh every 5 seconds
+        setUpdater(setInterval(() => refresh(), 5000));
+        refresh(); // refresh right away
+        return () => clearInterval(updater); // have component run this after unmounting as cleanup
     }, []);
 
+    useEffect(() => {
+        clearInterval(updater); // clear previous refresher (which was for a diff convo)
+        // refresh every 5 seconds
+        setUpdater(setInterval(() => refresh(), 5000));
+        refresh(); // refresh right away
+        return () => clearInterval(updater); // have component run this after unmounting as cleanup
+    }, conversationId);
+
```

Comment:
```suggestion
    useEffect(() => {
        clearInterval(updater); // clear previous refresher (which was for a diff convo)
        // refresh every 5 seconds
        setUpdater(setInterval(() => refresh(), 5000));
        refresh(); // refresh right away
        return () => clearInterval(updater); // have component run this after unmounting as cleanup
    }, [conversationId]);

```

This seems to fix the chat switch for me